include/picongpu/particles/ionization/byField/IonizationCurrent/JIonizationCalc.hpp

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2020 Jakob Trojok
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+namespace picongpu
+{
+namespace particles
+{
+namespace ionization
+{
+    /** calculates ionization current
+     */
+    struct JIonizationCalc
+    {
+        /** functor calculating ionization current
+         */
+        HDINLINE float3_X
+        operator()( float_X const ionizationEnergy, float3_X const eField )
+        {
+            float3_X jion = ionizationEnergy * eField / math::abs(eField) / math::abs(eField) / DELTA_T / ::picongpu::CELL_VOLUME;
```

Comment:
There should be `math::abs2`, which would be much less computing for this expression.